handler/src/main/java/io/netty/handler/ssl/SslContext.java

```diff
@@ -1010,13 +1010,24 @@ protected static TrustManagerFactory buildTrustManagerFactory(
         CertificateFactory cf = CertificateFactory.getInstance("X.509");
         X509Certificate[] x509Certs = new X509Certificate[certs.length];
 
+        int i = 0;
         try {
-            for (int i = 0; i < certs.length; i++) {
-                x509Certs[i] = (X509Certificate) cf.generateCertificate(new ByteBufInputStream(certs[i]));
+            for (; i < certs.length; i++) {
+                InputStream is = new ByteBufInputStream(certs[i], true);
+                try {
+                    x509Certs[i] = (X509Certificate) cf.generateCertificate(is);
+                } finally {
+                    try {
+                        is.close();
+                    } catch (IOException e) {
+                        // This is not expected to happen, but re-throw in case it does.
+                        throw new RuntimeException(e);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Refactor this code to not throw exceptions in finally blocks. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS1163)
![BLOCKER](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png) Remove this throw statement from this finally block. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS1143)